### DISCUSSION TITLE
Usage with Kubernetes proxy lead to problem with path name

### DISCUSSION THREAD

**Discussion by Panzki** [nodejs/undici#1024] [Q&A]
Hi everyone,
I', using the [HTTP datasource](https://github.com/StarpTech/apollo-datasource-http) from the Apollo GraphQL server project, which makes use of undici. For local development I wanted to point my data source into my Kubernetes cluster using the Kubernetes [proxy feature](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#manually-constructing-apiserver-proxy-urls). To do so one has to create a URL that contains information for the Kubernetes proxy, how to forward requests to that URL. 

So my URL looks like this: `http://localhost:8001/api/v1/namespaces/some-namespace/services/some-service:8080/proxy/`. Sadly this leads to an exception in undici when parsing the origin.
https://github.com/nodejs/undici/blob/727ba62fa05b97b5cfd380b92cf94f7043de8d05/lib/core/util.js#L71-L79
Since the pathname of my URL is `/api/v1/namespaces/some-namespace/services/some-service:8080/proxy/` I get an error and can not continue.

Could you explain, why you consider an URL that has a path name other than `/` as invalid?

Thanks

**Comment by mcollina** [comment]
You would need to specify `http://localhost:8001/` as origin and the rest as the path.
